diff --git a/discover/DiscoverMainWindow.cpp b/discover/DiscoverMainWindow.cpp
index b4cbe68..f8e395e 100644
--- a/discover/DiscoverMainWindow.cpp
+++ b/discover/DiscoverMainWindow.cpp
@@ -233,6 +233,7 @@ void DiscoverMainWindow::integrateObject(QObject* object)
     }
 
     object->installEventFilter(this);
+    connect(object, &QObject::destroyed, qGuiApp, &QCoreApplication::quit);
 }
 
 bool DiscoverMainWindow::eventFilter(QObject * object, QEvent * event)
diff --git a/discover/main.cpp b/discover/main.cpp
index 6f41f91..5ad6320 100644
--- a/discover/main.cpp
+++ b/discover/main.cpp
@@ -122,6 +122,9 @@ int main(int argc, char** argv)
         mainWindow = new DiscoverMainWindow(s_decodeCompactMode->value(parser->value(QStringLiteral("compact")), DiscoverMainWindow::Full));
         QObject::connect(&app, &QCoreApplication::aboutToQuit, mainWindow, &DiscoverMainWindow::deleteLater);
         QObject::connect(service, &KDBusService::activateRequested, mainWindow, [mainWindow](const QStringList &arguments, const QString &/*workingDirectory*/){
+            if (!mainWindow->rootObject())
+                QCoreApplication::instance()->quit();
+
             mainWindow->rootObject()->raise();
             if (arguments.isEmpty())
                 return;
diff --git a/libdiscover/Category/Category.h b/libdiscover/Category/Category.h
index f947323..cc7355f 100644
--- a/libdiscover/Category/Category.h
+++ b/libdiscover/Category/Category.h
@@ -87,6 +87,5 @@ private:
     QSet<QString> m_plugins;
     bool m_isAddons = false;
 };
-Q_DECLARE_METATYPE(QList<Category *>)
 
 #endif
diff --git a/libdiscover/backends/KNSBackend/KNSBackend.cpp b/libdiscover/backends/KNSBackend/KNSBackend.cpp
index 971e7b1..0ea70dc 100644
--- a/libdiscover/backends/KNSBackend/KNSBackend.cpp
+++ b/libdiscover/backends/KNSBackend/KNSBackend.cpp
@@ -309,9 +309,7 @@ static ResultsStream* voidStream()
 
 ResultsStream* KNSBackend::search(const AbstractResourcesBackend::Filters& filter)
 {
-    Q_ASSERT(isValid());
-
-    if ((!filter.resourceUrl.isEmpty() && filter.resourceUrl.scheme() != QLatin1String("kns")) || !filter.mimetype.isEmpty())
+    if (!m_isValid || (!filter.resourceUrl.isEmpty() && filter.resourceUrl.scheme() != QLatin1String("kns")) || !filter.mimetype.isEmpty())
         return voidStream();
 
     if (filter.resourceUrl.scheme() == QLatin1String("kns")) {
diff --git a/libdiscover/backends/PackageKitBackend/PackageKitBackend.cpp b/libdiscover/backends/PackageKitBackend/PackageKitBackend.cpp
index 472fd90..2f520ed 100644
--- a/libdiscover/backends/PackageKitBackend/PackageKitBackend.cpp
+++ b/libdiscover/backends/PackageKitBackend/PackageKitBackend.cpp
@@ -269,6 +269,9 @@ void PackageKitBackend::resolvePackages(const QStringList &packageNames)
 
 void PackageKitBackend::fetchUpdates()
 {
+    if (m_updater->isProgressing())
+        return;
+
     PackageKit::Transaction * tUpdates = PackageKit::Daemon::getUpdates();
     connect(tUpdates, &PackageKit::Transaction::finished, this, &PackageKitBackend::getUpdatesFinished);
     connect(tUpdates, &PackageKit::Transaction::package, this, &PackageKitBackend::addPackageToUpdate);
diff --git a/libdiscover/backends/PackageKitBackend/PackageKitNotifier.cpp b/libdiscover/backends/PackageKitBackend/PackageKitNotifier.cpp
index 5ce030f..97f083e 100644
--- a/libdiscover/backends/PackageKitBackend/PackageKitNotifier.cpp
+++ b/libdiscover/backends/PackageKitBackend/PackageKitNotifier.cpp
@@ -31,7 +31,6 @@
 
 PackageKitNotifier::PackageKitNotifier(QObject* parent)
     : BackendNotifierModule(parent)
-    , m_update(NoUpdate)
     , m_securityUpdates(0)
     , m_normalUpdates(0)
 {
@@ -75,7 +74,6 @@ void PackageKitNotifier::recheckSystemUpdateNeeded()
         PackageKit::Transaction * trans = PackageKit::Daemon::getUpdates();
         trans->setProperty("normalUpdates", 0);
         trans->setProperty("securityUpdates", 0);
-        trans->setProperty("update", NoUpdate);
         connect(trans, &PackageKit::Transaction::package, this, &PackageKitNotifier::package);
         connect(trans, &PackageKit::Transaction::finished, this, &PackageKitNotifier::finished);
     }
@@ -89,11 +87,9 @@ void PackageKitNotifier::package(PackageKit::Transaction::Info info, const QStri
         case PackageKit::Transaction::InfoBlocked:
             break; //skip, we ignore blocked updates
         case PackageKit::Transaction::InfoSecurity:
-            trans->setProperty("update", qMax<int>(Security, trans->property("update").toInt()));
             trans->setProperty("securityUpdates", trans->property("securityUpdates").toInt()+1);
             break;
         default:
-            trans->setProperty("update", qMax<int>(Normal, trans->property("update").toInt()));
             trans->setProperty("normalUpdates", trans->property("normalUpdates").toInt()+1);
             break;
     }
@@ -105,13 +101,10 @@ void PackageKitNotifier::finished(PackageKit::Transaction::Exit /*exit*/, uint)
 
     const uint normalUpdates = trans->property("normalUpdates").toInt();
     const uint securityUpdates = trans->property("securityUpdates").toInt();
-    const Update update = Update(trans->property("update").toInt());
-
-    const bool changed = update != m_update || normalUpdates != m_normalUpdates || securityUpdates != m_securityUpdates;
+    const bool changed = normalUpdates != m_normalUpdates || securityUpdates != m_securityUpdates;
 
     m_normalUpdates = normalUpdates;
     m_securityUpdates = securityUpdates;
-    m_update = update;
 
     if (changed) {
         Q_EMIT foundUpdates();
@@ -120,7 +113,7 @@ void PackageKitNotifier::finished(PackageKit::Transaction::Exit /*exit*/, uint)
 
 bool PackageKitNotifier::isSystemUpToDate() const
 {
-    return m_update == NoUpdate;
+    return m_securityUpdates == 0 && m_normalUpdates == 0;
 }
 
 uint PackageKitNotifier::securityUpdatesCount()
diff --git a/libdiscover/backends/PackageKitBackend/PackageKitNotifier.h b/libdiscover/backends/PackageKitBackend/PackageKitNotifier.h
index ddd3485..1fead43 100644
--- a/libdiscover/backends/PackageKitBackend/PackageKitNotifier.h
+++ b/libdiscover/backends/PackageKitBackend/PackageKitNotifier.h
@@ -34,12 +34,6 @@ Q_OBJECT
 Q_PLUGIN_METADATA(IID "org.kde.discover.BackendNotifierModule")
 Q_INTERFACES(BackendNotifierModule)
 public:
-    enum Update {
-        NoUpdate,
-        Security,
-        Normal
-    };
-    Q_ENUM(Update)
     explicit PackageKitNotifier(QObject* parent = nullptr);
     ~PackageKitNotifier() override;
 
@@ -56,7 +50,6 @@ private Q_SLOTS:
 private:
     QProcess* checkAptVariable(const QString &aptconfig, const QLatin1String& varname, std::function<void(const QStringRef& val)> func);
 
-    Update m_update;
     uint m_securityUpdates;
     uint m_normalUpdates;
     QPointer<PackageKit::Transaction> m_refresher;
diff --git a/notifier/DiscoverNotifier.cpp b/notifier/DiscoverNotifier.cpp
index b2ccce3..35f8261 100644
--- a/notifier/DiscoverNotifier.cpp
+++ b/notifier/DiscoverNotifier.cpp
@@ -26,7 +26,6 @@
 #include <KLocalizedString>
 #include <KNotification>
 #include <KPluginFactory>
-#include <QMenu>
 
 DiscoverNotifier::DiscoverNotifier(QObject * parent)
     : QObject(parent)
@@ -40,7 +39,7 @@ DiscoverNotifier::DiscoverNotifier(QObject * parent)
     }
     connect(&m_timer, &QTimer::timeout, this, &DiscoverNotifier::showUpdatesNotification);
     m_timer.setSingleShot(true);
-    m_timer.setInterval(180000);
+    m_timer.setInterval(1000);
     updateStatusNotifier();
 }
 
diff --git a/notifier/plasmoid/metadata.desktop b/notifier/plasmoid/metadata.desktop
index 348f258..e46ab5e 100644
--- a/notifier/plasmoid/metadata.desktop
+++ b/notifier/plasmoid/metadata.desktop
@@ -90,7 +90,7 @@ X-KDE-PluginInfo-Author=The Plasma Team
 X-KDE-PluginInfo-Email=plasma-devel@kde.org
 X-KDE-PluginInfo-Name=org.kde.discovernotifier
 X-KDE-PluginInfo-Version=3.0
-X-KDE-PluginInfo-Website=http://plasma.kde.org/
+X-KDE-PluginInfo-Website=https://www.kde.org/plasma-desktop
 X-KDE-PluginInfo-Category=Online Services
 X-KDE-PluginInfo-Depends=
 X-KDE-PluginInfo-License=GPLv2+
